modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java

```diff
@@ -752,10 +752,18 @@ public String toEnumValue(String value, String datatype) {
                 "int".equalsIgnoreCase(datatype)) {
             return value;
         } else {
-            return "'" + escapeText(value) + "'";
+            return "'" + escapeTextInSingleQuotes(value) + "'";
         }
     }
 
+    public String escapeTextInSingleQuotes(String input) {
+        if (input == null) {
+            return null;
+        }
+
+        return escapeText(input).replace("'", "\\'");
+    }
+
```

Comment:
thanks for the PR.

what about putting this method in default codegen instead?

minor suggestion: add a docstring or 1-liner explaining what this function does

Reply:
Sure, I've done both of these